examples/command.rs

```diff
@@ -0,0 +1,642 @@
+#![allow(clippy::type_complexity)]
+
+use std::ops::DerefMut;
+
+use command::graph::CommandGraphBuilder;
+use command::handler::CommandResultEvent;
+use command::parsers::entity_selector::{EntitySelector, EntitySelectors};
+use command::parsers::{CommandArg, GreedyString, QuotableString};
+use command::scopes::CommandScopes;
+use command::{parsers, Command, CommandApp, CommandScopeRegistry, ModifierValue};
+use command_macros::Command;
+use parsers::{Vec2 as Vec2Parser, Vec3 as Vec3Parser};
+use rand::prelude::IteratorRandom;
+use valence::entity::living::LivingEntity;
+use valence::prelude::*;
+use valence::*;
+use valence_server::op_level::OpLevel;
+
+const SPAWN_Y: i32 = 64;
+
+#[derive(Command, Debug, Clone)]
+#[paths("teleport", "tp")]
+#[scopes("valence.command.teleport")]
+enum Teleport {
+    #[paths = "{location}"]
+    ExecutorToLocation { location: Vec3Parser },
+    #[paths = "{target}"]
+    ExecutorToTarget { target: EntitySelector },
+    #[paths = "{from} {to}"]
+    TargetToTarget {
+        from: EntitySelector,
+        to: EntitySelector,
+    },
+    #[paths = "{target} {location}"]
+    TargetToLocation {
+        target: EntitySelector,
+        location: Vec3Parser,
+    },
+}
+
+#[derive(Command, Debug, Clone)]
+#[paths("gamemode", "gm")]
+#[scopes("valence.command.gamemode")]
+enum Gamemode {
+    #[paths("survival {target?}", "{/} gms {target?}")]
+    Survival { target: Option<EntitySelector> },
+    #[paths("creative {target?}", "{/} gmc {target?}")]
+    Creative { target: Option<EntitySelector> },
+    #[paths("adventure {target?}", "{/} gma {target?}")]
+    Adventure { target: Option<EntitySelector> },
+    #[paths("spectator {target?}", "{/} gmspec {target?}")]
+    Spectator { target: Option<EntitySelector> },
+}
```

Comment:
I tried to make a basic version of the gamemode command and wrote this.

```rust
#[derive(Command, Debug, Clone)]
#[paths("gamemode {gamemode} {target?}")]
struct GameModeCommand2 {
    gamemode: GameMode,
    target: Option<EntitySelector>,
}
```

But it doesn't compile because only `enum` is allowed. So I had to write this instead.

```rust
#[derive(Command, Debug, Clone)]
#[paths("gamemode")]
enum GameModeCommand {
    #[paths("{gamemode} {target?}")]
    GameMode {
        gamemode: GameMode,
        target: Option<EntitySelector>,
    }
}
```

Can we change the macro to support `struct` commands?